### PR TITLE
Cherry pick commits of 0.4.x

### PR DESCRIPTION
**Pull request by QilongZhang**
cherry-pick commits of 0.4.x

### CODE REVIEW

File: sofa-ark-parent/core/common/src/main/java/com/alipay/sofa/ark/common/util/StringUtils.java [hunk 1]
```diff
@@ -90,4 +93,31 @@ public static String listToStr(Set<String> stringSet, String delimiter) {
...
     }
+
+    /**
+     * Check whether start with anotherString when transformed to lower case.
+     *
+     * @param thisString
+     * @param anotherString
+     * @return
+     */
+    public static boolean startWithToLowerCase(String thisString, String anotherString) {
```
Comment (ujjboy) [inline sofa-ark-parent/core/common/src/main/java/com/alipay/sofa/ark/common/util/StringUtils.java]: This is a new feature and should be commit in another PR.

**Comment by QilongZhang** [comment]
@ujjboy
